lib/bundler/rubygems_integration.rb

```diff
@@ -323,11 +323,12 @@ def reverse_rubygems_kernel_mixin
     def replace_gem(specs, specs_by_name)
       reverse_rubygems_kernel_mixin
 
-      executables = specs.map(&:executables).flatten
+      executables = nil
 
       kernel = (class << ::Kernel; self; end)
       [kernel, ::Kernel].each do |kernel_class|
         redefine_method(kernel_class, :gem) do |dep, *reqs|
+          executables ||= specs.map(&:executables).flatten
```

Comment:
I'm not happy with this change, I'd like to investigate if this check is even necessary. It was introduced 7 years ago without any test coverage...